File: src/bytecode/structs/Local.ts
import { MemoryStream } from "../../util/MemoryStream";

export class Local {
  name: string;
  start: number;
  end: number;

  constructor(byteStream: MemoryStream) {
    this.name = byteStream.readString();
    this.start = byteStream.readInt();
    this.end = byteStream.readInt();
  }
}

File: src/bytecode/structs/local.rs
use crate::util::read_stream::ReadStream;

pub struct Local {
    pub name: String,
    pub start: u64,
    pub end: u64,
}

impl Local {
    pub fn new(memory_stream: &mut ReadStream) -> Self {
        Self {
            name: memory_stream.read_string(),
            start: memory_stream.read_int(),
            end: memory_stream.read_int(),
        }
    }
}
